Classes/Command/FluidPrecompileCommandController.php

```diff
@@ -0,0 +1,220 @@
+<?php
+namespace NamelessCoder\CmsFluidPrecompilerModule\Command;
+
+use NamelessCoder\CmsFluidPrecompilerModule\Service\FluidPrecompilerService;
+use Symfony\Component\Console\Helper\TableSeparator;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Mvc\Controller\CommandController;
+use TYPO3\CMS\Extbase\Object\ObjectManager;
+use TYPO3Fluid\Fluid\Core\Cache\FluidCacheWarmupResult;
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+
+/**
+ * Class FluidPrecompileCommandController
+ */
+class FluidPrecompileCommandController extends CommandController
```

Comment:
I suggest making this a Symfony command instead, this way we skip the whole Extbase overhead. The command can be used only with TYPO3v8 or newer anyways.

Reply:
Good point, let's do that.

Reply:
I'd normally agree with you on this.
I prefer a Symfony Command, too, because TYPO3's Command API is very limited.

However I wanted to stay compatible with the TYPO3 Core (or Extbase) and especially the [TYPO3 Console](https://github.com/TYPO3-Console/TYPO3-Console).

Another benefit of the Command Controller is, that it can be used by the scheduler extension.

A Symfony Command would also require some sort of Symfony Application bootstrapping (even if its a oneliner). How would you've done this? Creating an extra _bin_ script?

I'd suggest to add an additional Symfony Command but also keep the TYPO3 Command.

Reply:
TYPO3 [natively supports](https://docs.typo3.org/typo3cms/extensions/core/8.7/singlehtml/Index.html#feature-73042-introduce-native-support-for-symfony-console) and uses Symfony commands. And starting from TYPO3v9 you have a task for executing registered Symfony tasks.

TL;DR: it's easier than you think. ;-)

Reply:
Okay, I didn't know. Sadly, the core documentation on this very poor and as far as I've checked, the extbase commands rely on the TYPO3 Command.

I'm going to migrate the CommandController to a Symfony Command.